### ISSUE TITLE
Could not lex JSON (with Unicode escapes + Pygments 2.7.0)

### ISSUE THREAD

**Issue by vytas7** [sphinx-doc/sphinx#8210]
**Describe the bug**
Sphinx fails to lex a JSON source code block in case the JSON contains Unicode escapes, using the newest (at the time of writing) `Pygments==2.7.0`:
```
WARNING: Could not lex literal_block as "json". Highlighting skipped.
```
`Pygments==2.6.1` works as expected. However, I've tested highlighting manually with Pygments==2.7.0 using the coloured terminal output formatter, and it seemed to work.

**To Reproduce**
```
$ pip install Sphinx==3.2.1 Pygments==2.7.0
$ touch conf.py
$ echo -e 'Is something wrong?\n===================\n\n.. sourcecode:: json\n\n   {\n     "French": "Fran\\u00e7ais",\n     "bug": true\n   }\n' > index.rst
$ sphinx-build . build
```

**Expected behavior**
JSON is lexed and highlighted; no warnings are emitted.

**Environment info**
- OS: Ubuntu 20.04.1 LTS
- Python version: 3.8.2
- Sphinx version: 3.2.1
- Sphinx extensions:
- Extra tools:

**Comment by tk0miya** [comment]
It seems pygments-2.7.0 raises an error for the JSON data.
```
$ echo '{\n     "French": "Fran\\u00e7ais",\n     "bug": true\n   }\n' | pygmentize -l json -F raiseonerror

*** Error while highlighting:
pygments.filters.ErrorToken: "
   (file "/Users/tkomiya/.pyenv/versions/3.8.5/lib/python3.8/site-packages/pygments/filters/__init__.py", line 783, in filter)
*** If this is a bug you want to report, please rerun with -v.
{
```

I don't know this is an expected change of pygments. Please ask this to their project.

Anyway, you can avoid the error by `:force:` option of `code-block` directive.

**Comment by vytas7** [comment]
Thanks for looking into this @tk0miya !
I was invoking `pygments` programmatically and it worked. I can indeed reproduce your error using `pygmentize` from the command line.

I'll file a `Pygments` bug. **Edit**: filed https://github.com/pygments/pygments/issues/1544

**Comment by vytas7** [comment]
Closing this as the newly released `Pygments==2.7.1` fix addresses the issue.
